### ISSUE TITLE
Hanging indentation is a little odd for callables that return context managers

### ISSUE THREAD

**Issue by mgilson** [PyCQA/pylint#963]
I don't think that this is necessarily a bug -- More a consideration for a feature request.  I think that indentation rules should be determined based on whether the statement is starting a suite.  `if` suites seem to work pretty much how I would expect, but context managers don't.  It's a tricky thing to describe, so rather I'll provide an example script with the way the linter currently wants it to be formatted and with how I would expect it to be formatted (at least optionally):
### Current Behavior

The following script lints fine:

``` py
"""module docstring"""
import os
import mock

FOO = 1
BAR = 2
if (FOO and
        BAR):
    pass

if (os.path.
        abspath('foo')):
    pass

if (os.path.abspath(
        'foo')):
    pass

if os.path.abspath(
        'foo'):
    pass

with mock.patch.object(
    os.path,
    'abspath',
    return_value='whatever'):
    pass
```
### Expected behavior

I would have expected that there would be a lint error on the second line of the `with` statement.  e.g., it seems like to be consistent with the `if` suite, we'd want the context manager to be formatted as:

``` py
"""module docstring"""
import os
import mock

FOO = 1
BAR = 2
if (FOO and
        BAR):
    pass

if (os.path.
        abspath('foo')):
    pass

if (os.path.abspath(
        'foo')):
    pass

if os.path.abspath(
        'foo'):
    pass

with mock.patch.object(
    os.path,
    'abspath',
    return_value='whatever'):
    pass
```
### pylint --version output

```
No config file found, using default configuration
pylint 1.5.6, 
astroid 1.4.6
Python 2.7.10 (default, Oct 23 2015, 19:19:21) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]
```

**Comment by PCManticore** [comment]
Thank you, this definitely should be improved.

**Comment by butla** [comment]
@mgilson I don't see any difference in the two blocks of code.
I've stumbled on this issue because I'm checking if someone had the problem I have. Which I guess you do?

Anyways, according to [pep8](https://www.python.org/dev/peps/pep-0008/#indentation) hanging indent should show logic separation. That's why the style enforced by pylint:
```python
with whatever(
    bla,
    ble) as something:
    some_action()
```

should be replaced by 
```python
with whatever(
        bla,
        ble) as something:
    some_action()
```

@mgilson That's what you had in mind as well?

**Comment by mgilson** [comment]
Yes, I think that is what I had in mind... I'm not sure how I managed to write the wrong thing in the initial description.

**Comment by PCManticore** [comment]
I can no longer reproduce this with the latest pylint:
```
a.py:24:0: C0330: Wrong hanging indentation before block (add 4 spaces).
    os.pathcemareesteaceastafunctieiasavedemcumoreformateziacuma,
    ^   | (bad-continuation)
a.py:25:0: C0330: Wrong hanging indentation before block (add 4 spaces).
    'abspath',
    ^   | (bad-continuation)
a.py:26:0: C0330: Wrong hanging indentation before block (add 4 spaces).
    return_value='whatever'):
    ^   | (bad-continuation)
```

**Comment by butla** [comment]
@PCManticore Yep, it seems to have been fixed some time ago.
